apps/OpenSpace/ext/launcher/src/launcherwindow.cpp

```diff
@@ -637,20 +678,117 @@ void LauncherWindow::openProfileEditor(const std::string& profile, bool isUserPr
+    sgct::config::Cluster preview;
+    if (winCfg.empty()) {
+        SgctEdit editor(this, _userConfigPath);
+        ret = editor.exec();
+        if (ret == QDialog::DialogCode::Accepted) {
+            handleReturnFromWindowEditor(
+                editor.cluster(),
+                editor.saveFilename(),
+                saveWindowCfgPath
+            );
+        }
+    }
+    else {
+        try {
+            sgct::config::GeneratorVersion previewGenVersion =
+                sgct::readConfigGenerator(winCfg);
+            sgct::loadFileAndSchemaThenValidate(
+                winCfg,
+                _configPath + "/schema/sgct.schema.json",
+                "This configuration file is unable to generate a proper display"
+            );
+            sgct::loadFileAndSchemaThenValidate(
+                winCfg,
+                _configPath + "/schema/sgcteditor.schema.json",
+                "This configuration file is valid for generating a display, but "
+                "its format does not match the window editor requirements and "
+                "cannot be opened in the editor"
+            );
+            if (previewGenVersion.versionCheck(minimumVersion)) {
+                try {
+                    preview = sgct::readConfig(
+                        winCfg,
+                        "This configuration file is unable to generate a proper display "
+                        "due to a problem detected in the readConfig function"
+                    );
+                }
+                catch (std::runtime_error& e) {
+                    //Re-throw an SGCT error exception with the runtime exception message
+                    throw sgct::Error(
+                        sgct::Error::Component::ReadConfig,
+                        6082,
+                        fmt::format(
+                            "Importing of this configuration file failed because of a "
+                            "problem detected in the readConfig function:\n\n{}",
+                            e.what()
+                        )
+                    );
+                }
+                SgctEdit editor(
+                    preview,
+                    winCfg,
+                    saveWindowCfgPath,
+                    _readOnlyWindowConfigs,
+                    this
+                );
+                ret = editor.exec();
+                if (ret == QDialog::DialogCode::Accepted) {
+                    handleReturnFromWindowEditor(
+                        editor.cluster(),
+                        editor.saveFilename(),
+                        saveWindowCfgPath
+                    );
+                }
+            }
+            else {
+                editRefusalDialog(
+                    "File Format Version Error",
+                    fmt::format(
+                        "File '{}' does not meet the minimum required version of {}.",
+                        winCfg,
+                        minimumVersion.versionString()
```

Comment:
This line can be on the previous one without a line break needed (https://github.com/OpenSpace/OpenSpace/wiki/Coding-Style#handling-line-breaks)